source/web-service/flaskapp/routes/records.py

```diff
@@ -540,6 +549,72 @@ def entity_record(entity_id):
             return abort(response)
 
 
+# 'DELETE' method.
+@records.route("/<path:id>", methods=["DELETE"])
+def delete(id):
+
+    result_dict = {}
+
+    # Authentication
+    status = authenticate_bearer(request)
+    if status != status_ok:
+        response = construct_error_response(status)
+        return abort(response)
+
+    current_app.logger.debug("Authentication checked - DELETE request allowed.")
+
+    # Get record from DB
+    db_rec = get_record(id)
+
+    # No such record. Return {"id": None} (format identical to 'ingest/delete')
+    if db_rec is None:
+        result_dict[id] = "null"
+        return jsonify(result_dict), 200
+
+    # Process DELETE
+    with db.session.no_autoflush:
+        try:
+            process_activity(db_rec.id, Event.Delete)
+            record_delete(db_rec, None, True)
```

Comment:
Should be `record_delete(db_rec, None)` - don't commit at this stage.

Reply:
Not sure why this comment is hidden by github - replying to it in hope it pops back up!